From acca97d286fadfd2801a4aad55021f9881f87af3 Mon Sep 17 00:00:00 2001
From: aken_liu <aken_liu@accton.com.tw>
Date: Mon, 10 Jun 2019 14:42:58 +0800
Subject: [PATCH] ACCTON-996: Too many open files in system Root Casue: After
 fan controller opens and reads the thermal sensor's sysfs, we do not close
 it. The fan controller is a periodic task, it casues there are a lot of
 opened files in the linux system. Solution: After reading the data, we shall
 close the file.

---
 drivers/hwmon/accton_t600_fan.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/hwmon/accton_t600_fan.c b/drivers/hwmon/accton_t600_fan.c
index 2eef392..8c4046d 100644
--- a/drivers/hwmon/accton_t600_fan.c
+++ b/drivers/hwmon/accton_t600_fan.c
@@ -903,6 +903,7 @@ static int read_file_contents(char *path, char *buf, long data_len, struct devic
 		}
 
 		kernel_read(fp, 0, buf, data_len);
+                filp_close(fp, NULL);
 		break;
 	}
 
-- 
1.9.1

